diff --git a/distrobox-init b/distrobox-init
index f2153b9..8a3dd7e 100755
--- a/distrobox-init
+++ b/distrobox-init
@@ -1567,7 +1567,9 @@ if [ "${nvidia}" -eq 1 ]; then
 
 		if [ "${type}" = "link" ]; then
 			mkdir -p "$(dirname "${dest_file}")"
-			cp -d "${nvidia_lib}" "${dest_file}"
+			if [[ $(md5sum "${nvidia_lib}" | awk '{print $1}') != $(md5sum "${dest_file}" | awk '{print $1}') ]]; then
+				cp -d "${nvidia_lib}" "${dest_file}"
+			fi
 			continue
 		fi
 
